### ISSUE TITLE
Needs better documentation

### ISSUE THREAD

**Issue by mctrivia** [uNetworking/uWebSockets#1464]
I have been having a lot of difficulty trying to get a simple http api server built in to my application. It will be communicating only with only other servers on the same local network so SSL is not needed.

I tried setting up uWebSockets but the documentation is not very good. I did manage to get the IDE to say all errors where good by using the command:

`vcpkg install libuv uwebsockets --triplet=x64-windows`

and using the following CMakeLists.txt

```
cmake_minimum_required(VERSION 3.21) project(exchange)

set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules) set(CMAKE_CXX_STANDARD 20) set(CMAKE_C_FLAGS "-lgmp") set(ZLIB_DEPS_DIR /cygdrive/c/src/vcpkg/packages/zlib_x64-windows/bin)

set(HEADER_FILES
    long list of .h files )

set(SOURCE_FILES
    long list of .cpp files )

find_package(GMP REQUIRED)

include_directories(uWebSockets) add_executable(exchange main.cpp  ${SOURCE_FILES} ${HEADER_FILES}) target_link_libraries(exchange gmp gmpxx z uv)


set(UWEBSOCKETS_INCLUDE_DIRS /cygdrive/c/src/vcpkg/packages/uwebsockets_x64-windows/include/uwebsockets) set(ZLIB_INCLUDE_DIRS /cygdrive/c/src/vcpkg/packages/zlib_x64-windows/include) set(USOCKETS_INCLUDE_DIRS /cygdrive/c/src/vcpkg/packages/usockets_x64-windows/include)

target_include_directories(exchange PRIVATE ${ZLIB_INCLUDE_DIRS}) target_include_directories(exchange PRIVATE ${UWEBSOCKETS_INCLUDE_DIRS}) target_include_directories(exchange PRIVATE ${USOCKETS_INCLUDE_DIRS})




add_library(exchange_lib STATIC ${SOURCE_FILES} ${HEADER_FILES})
```
But i keep getting the error when I try to build it

`/cygdrive/c/src/vcpkg/packages/uwebsockets_x64-windows/include/uwebsockets/Loop.h:31: undefined reference to `us_loop_ext'`
What am I missing?

**Comment by alexhultman** [comment]
I don't really think this is any particular issue with the library, more so an issue with your setup consisting of:

* Windows
* CMake
* vcpkg
* Cygwin

None of those are supported in any way. We prefer regular macOS or Linux, on which it just works out of the box. Also, I'm pretty sure vcpkg does not affect your Cygwin universe.

That's a linker error, your linker is not linking against uSockets. I don't know how your setup is, and I cannot possibly support all combinations of all kinds of set ups, this project is just regular standard C++ and standard C, how you built it on whatever platform is up to you.

That said, on macOS and Linux it builds without any problems by just hitting "make".

**Comment by ariestav** [comment]
@alexhultman I thought windows was supported.  I'm able to build by linking to it on Visual Studio 2022.  Not sure if this is accurate, but I followed these steps and the Visual Studio IDE recognizes the symbols.  In the linker I am linking to uSockets.lib that vcpkg installed.  I'm building a .dll for a host app with this, but when the built .dll is loaded at the host app's runtime, the host app refuses to load it.  When I remove the code for uSockets, it loads into the host app fine.  Any ideas?  If we are linking against uSocket.lib, we don't need the uSockets.dll accessible for runtime, correct?  Isn't the uSockets.lib file a static lib?

**Comment by alexhultman** [comment]
Windows is supported yes, correction.

**Comment by ariestav** [comment]
@alexhultman Thanks for the response.  Is there a guide out there for using this library in Visual Studio somewhere?  I found [this post on StackOverflow](https://stackoverflow.com/questions/61833538/how-to-use-uwebsockets-from-visual-studio-c), but do the steps seem accurate?

**Comment by ariestav** [comment]
Okay, I figured out the issue on Windows.  I had to drop the following built `.dll` files into the folder that contains the host app executable.  

```
zlib1.dll
uSockets.dll
libuv.dll
```

This host app executable loads the `.dll` file I'm building with uWebSockets.  Is there any way to set up the compiler so that the built output binary does not require these `.dll` files to exist in order to execute at runtime?
